src/components/InventoryV2/view/BatchDetailsModalForNursery.tsx

```diff
@@ -0,0 +1,544 @@
+
+  const MANDATORY_FIELDS = [
+    'facilityId',
+    'germinatingQuantity',
+    'notReadyQuantity',
+    'readyQuantity',
+    'addedDate',
+  ] as const;
+  type MandatoryField = (typeof MANDATORY_FIELDS)[number];
+
+  const hasErrors = () => {
+    if (record) {
+      return MANDATORY_FIELDS.some((field: MandatoryField) => record[field] === '' || record[field] === undefined);
+    }
+    return true;
+  };
+
+  const saveBatch = async () => {
+    if (record) {
+      if (hasErrors()) {
+        setValidateFields(true);
+        return;
+      }
+
+      let response: (Response & BatchData) | (Response & BatchId) | undefined;
+      let responseQuantities: Partial<Response> = { requestSucceeded: true, error: undefined };
+
+      if (record.id === -1) {
+        if (record.accessionId && isBatch(record)) {
+          const nurseryTransferRecord: NurseryTransfer = {
+            date: getTodaysDateFormatted(timeZone),
+            destinationFacilityId: record.facilityId,
+            germinatingQuantity: record.germinatingQuantity,
+            notReadyQuantity: record.notReadyQuantity,
+            notes: record.notes,
+            readyByDate: record.readyByDate,
+            readyQuantity: record.readyQuantity,
+            // TODO figure out if I need this or if is auto populated by auth
+            // withdrawnByUserId: 0
+          };
+          const accessionResponse = await AccessionService.transferToNursery(nurseryTransferRecord, record.accessionId);
+          if (!accessionResponse.requestSucceeded || !accessionResponse.data) {
+            snackbar.toastError(accessionResponse.error || strings.GENERIC_ERROR);
+            return;
+          }
+
+          record.id = accessionResponse.data.batch.id;
+          record.version = accessionResponse.data.batch.version;
+
+          // This is where the sub locations are associated to the batch created through the nursery transfer
+          response = await NurseryBatchService.updateBatch(record);
+          if (response.batch) {
+            responseQuantities = await NurseryBatchService.updateBatchQuantities({
+              ...record,
+              version: response.batch.version,
+            });
+          }
+        } else {
+          response = await NurseryBatchService.createBatch(record as CreateBatchRequestPayload);
+        }
+      } else if (isBatch(record)) {
+        response = await NurseryBatchService.updateBatch(record);
+        if (response.batch) {
+          responseQuantities = await NurseryBatchService.updateBatchQuantities({
+            ...record,
+            version: response.batch.version,
+          });
+        }
+      }
+
+      if (response && response.requestSucceeded && responseQuantities.requestSucceeded) {
+        reload();
+        onCloseHandler();
+      } else {
+        snackbar.toastError((response && response.error) || responseQuantities.error || strings.GENERIC_ERROR);
+      }
+    }
+  };
```

Comment:
a tad concerned this could be fragile.. if user closes the browser during the request, etc.. wondering if we need to think about a BE api that does all this..? To be fair we do have other places in the UI code that does some multi api flows such as creating/deleting sublocations as part of a facility create/edit.

Reply:
I asked @sgrimm about it being implemented in the current API but he mentioned I should just call the two APIs consecutively. It wouldn't be bulletproof, but I could add an `onunload` event handler to warn the user about leaving the page while the requests are in flight, what do you think? Not sure how well it will work with a SPA though... Alternatively I could move these to redux, that way they will execute as desired regardless of the user's navigation (unless they leave the site obviously)

Reply:
Gotcha. Maybe something to discuss as a team at some point. The `onunload` at a minimum sounds good to me. I can update other flows later.

Reply:
Alright I added a note to bring it up at the next round table. How's this for now? 

I had a 5 second timeout between server calls in this example below:
![2023-11-21 12 48 31](https://github.com/terraware/terraware-web/assets/9541815/34dae59f-f5f3-42c3-be37-cbaaa88c321f)

Reply:
This is great!! Thanks so much 🎉